resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/Config.scala

```diff
@@ -54,6 +54,12 @@ private[spark] object Config extends Logging {
       .checkValues(Set("Always", "Never", "IfNotPresent"))
       .createWithDefault("IfNotPresent")
 
+  val IMAGE_PULL_SECRET =
+    ConfigBuilder("spark.kubernetes.container.image.pullSecret")
```

Comment:
Based on a conversation with the node team, this is meant to be an array of possibly many secret names; since one could have multiple secrets specified for different registries and spread across multiple k8s secrets. 

So, I'd suggest we implement the full API - this should be capable of accepting a comma separated list of one or more secret names.

Reply:
Also should be named `image.pullSecrets` to stay consistent with the field we're setting on the pods.

Reply:
@foxish done